Sourcecode/private/mx/impl/PropertiesWriter.cpp

```diff
@@ -165,26 +169,59 @@ namespace mx
                 key->getAttributes()->number = core::StaffNumber{ staffIndex + 1 };
             }
             
-            key->getKeyChoice()->setChoice( core::KeyChoice::Choice::traditionalKey );
-            auto traditionalKey = key->getKeyChoice()->getTraditionalKey();
-            traditionalKey->getFifths()->setValue( core::FifthsValue{ inKeyData.fifths } );
-            
-            if ( inKeyData.cancel != 0 )
+            if (!inKeyData.nonTraditionalKey)
             {
-                traditionalKey->setHasCancel( true );
-                traditionalKey->getCancel()->setValue( core::FifthsValue{ inKeyData.cancel } );
-            }
-            
-            if( inKeyData.mode == api::KeyMode::major || inKeyData.mode == api::KeyMode::minor )
-            {
-                traditionalKey->setHasMode( true );
-                traditionalKey->getMode()->setValue( core::ModeValue{ inKeyData.mode == api::KeyMode::major ? core::ModeEnum::major : core::ModeEnum::minor } );
+                key->getKeyChoice()->setChoice(core::KeyChoice::Choice::traditionalKey);
+                auto traditionalKey = key->getKeyChoice()->getTraditionalKey();
+                traditionalKey->getFifths()->setValue(core::FifthsValue{ inKeyData.fifths });
+
+                if (inKeyData.cancel != 0)
+                {
+                    traditionalKey->setHasCancel(true);
+                    traditionalKey->getCancel()->setValue(core::FifthsValue{ inKeyData.cancel });
+                }
+
+                if (inKeyData.mode == api::KeyMode::major || inKeyData.mode == api::KeyMode::minor)
+                {
+                    traditionalKey->setHasMode(true);
+                    traditionalKey->getMode()->setValue(core::ModeValue{ inKeyData.mode == api::KeyMode::major ? core::ModeEnum::major : core::ModeEnum::minor });
+                }
+
+            } else {
+                key->getKeyChoice()->setChoice(core::KeyChoice::Choice::nonTraditionalKey);
+
+                for (const api::PitchData& p : inKeyData.nonTraditionalPitchData)
+                {
+                    core::NonTraditionalKeyPtr k = core::makeNonTraditionalKey();
+                    if (p.step == api::Step::count || p.step == api::Step::unspecified)
+                    {
+                        // Would you prefer throw? assert?
```

Comment:
Not sure what is your standard in terms of "bad things reporting". Continue, throw, assert, format c:/ ?

Reply:
I think just silently fall back to `Step::c`. If I remember right I only use exceptions/assertions for bugs in the library, not for bad input.